src/lib/containers/AccessApprovalsTable.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  AccessApprovalSearchRequest,
+  AccessApprovalSearchResponse,
+  AccessApprovalSearchResult,
+  AccessApprovalSearchSort,
+  AccessApprovalSortField,
+} from '../utils/synapseTypes'
+import { searchAccessApprovals } from '../utils/SynapseClient'
+import Typography from '../utils/typography/Typography'
+import IconSvg from './IconSvg'
+import UserCard from './UserCard'
+import { displayToast } from './ToastMessage'
+import { PRODUCTION_ENDPOINT_CONFIG } from '../utils/functions/getEndpoint'
+import { SynapseSpinner } from './LoadingScreen'
+import { SortDirection } from './FavoritesPage'
+
+export type AccessApprovalsTableProps = {
+  accessorId: string
+  accessRequirementId?: string
+}
+export const AccessApprovalsTable: React.FunctionComponent<
+  AccessApprovalsTableProps
+> = ({ accessorId, accessRequirementId }) => {
+  const { accessToken } = useSynapseContext()
+  const [searchResult, setSearchResult] =
+    useState<AccessApprovalSearchResponse>()
+  const [isLoading, setIsLoading] = useState(false)
+  const [isDescending, setIsDescending] = useState(false)
+  const [arList, setArList] = useState<AccessApprovalSearchResult[]>()
+
+  const getApprovals = async () => {
+    try {
+      const searchRequest: AccessApprovalSearchRequest = {
+        accessorId,
+        accessRequirementId,
+      }
+      const approvalResponse = await searchAccessApprovals(
+        searchRequest,
+        accessToken,
+      )
+      setSearchResult(approvalResponse)
+      setArList(approvalResponse.results)
+    } catch (err: any) {
+      displayToast(err.reason as string, 'danger')
+    }
+  }
+
+  const onLoadMore = async () => {
+    setIsLoading(true)
+    try {
+      const searchRequest: AccessApprovalSearchRequest = {
+        accessorId,
+        accessRequirementId,
+        nextPageToken: searchResult?.nextPageToken,
+      }
+      const approvalResponse = await searchAccessApprovals(
+        searchRequest,
+        accessToken,
+      )
+      setSearchResult(approvalResponse)
+      if (arList) {
+        setArList([...arList, ...approvalResponse.results])
+      }
+    } catch (err: any) {
+      displayToast(err.reason as string, 'danger')
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  useEffect(() => {
+    getApprovals()
+  }, [accessToken, accessorId, accessRequirementId])
+
+  const sortColumn = async (
+    field: AccessApprovalSortField,
+    isDescending: boolean,
+  ) => {
+    try {
+      const direction = isDescending
+        ? ('DESC' as SortDirection)
+        : ('ASC' as SortDirection)
+      const sort: AccessApprovalSearchSort[] = [
+        {
+          field,
+          direction,
+        },
+      ]
+      const searchRequest: AccessApprovalSearchRequest = {
+        accessorId,
+        accessRequirementId,
+        sort,
+      }
+      const sortRequest = await searchAccessApprovals(
+        searchRequest,
+        accessToken,
+      )
```

Comment:
No reason for sorted requests to be invoked differently than any other prop change.